src/bin/inx-chronicle/api/stardust/core/routes.rs

```diff
@@ -344,11 +346,41 @@ async fn transaction_included_block(
         .collection::<BlockCollection>()
         .get_block_for_transaction(&transaction_id)
         .await?
-        .ok_or(MissingError::NoResults)?;
+        .ok_or(MissingError::NoResults)?
+        .block;
 
     Ok(IotaRawResponse::Json(block.into()))
 }
 
+async fn included_block_metadata(
+    database: Extension<MongoDb>,
+    Path(transaction_id): Path<String>,
+) -> ApiResult<IotaResponse<BlockMetadataResponse>> {
+    let transaction_id = TransactionId::from_str(&transaction_id).map_err(RequestError::from)?;
+
+    let res = database
+        .collection::<BlockCollection>()
+        .get_block_metadata_for_transaction(&transaction_id)
+        .await?
+        .ok_or(MissingError::NoResults)?;
+    let block_id = res.block_id;
+    let metadata = res.metadata;
+
+    Ok(BlockMetadataResponse {
+        block_id: block_id.to_hex(),
+        parents: metadata.parents.iter().map(BlockId::to_hex).collect(),
+        is_solid: metadata.is_solid,
+        referenced_by_milestone_index: Some(*metadata.referenced_by_milestone_index),
+        milestone_index: Some(*metadata.milestone_index),
+        ledger_inclusion_state: Some(metadata.inclusion_state.into()),
+        conflict_reason: Some(metadata.conflict_reason as u8),
+        should_promote: Some(metadata.should_promote),
+        should_reattach: Some(metadata.should_reattach),
+        white_flag_index: Some(metadata.white_flag_index),
+    }
+    .into())
```

Comment:
Should we `impl From<(BlockId,BlockMetadata)> for BlockMetadataResponse`? That way `included_block_metadata` and `block_metadata` could share this code?

Reply:
good idea!